lib/Onyx.js

```diff
@@ -1025,17 +1028,27 @@ function initializeWithDefaultKeyStates() {
  * Storage.setItem() from Onyx.clear() will have already finished and the merged
  * value will be saved to storage after the default value.
  *
+ * You may pass in keyStatesToPreserve to preserve Onyx keys you don't want to clear.
+ *
+ * @param {Array} keyStatesToPreserve ONYXKEYS values you want to keep
  * @returns {Promise<void>}
  */
-function clear() {
+function clear(keyStatesToPreserve = []) {
     return getAllKeys()
         .then((keys) => {
+            const keyValuesToPreserve = [];
             _.each(keys, (key) => {
-                const resetValue = lodashGet(defaultKeyStates, key, null);
-                cache.set(key, resetValue);
-                notifySubscribersOnNextTick(key, resetValue);
+                if (_.contains(keyStatesToPreserve, key)) {
+                    keyValuesToPreserve.push([key, get(key)]);
+                } else {
+                    const resetValue = lodashGet(defaultKeyStates, key, null);
```

Comment:
Should we also add the defaultKeyStates into the `keyValuesToPreserve` array?

Reply:
I can't think of a good reason to remove them from storage without adding them back in like we are doing with the key/values we want to preserve.